python/mxnet/gluon/loss.py

```diff
@@ -696,3 +696,35 @@ def hybrid_forward(self, F, pred, positive, negative):
                      axis=self._batch_axis, exclude=True)
         loss = F.relu(loss + self._margin)
         return _apply_weighting(F, loss, self._weight, None)
+
+
+class CenterLoss(Loss):
+    r""" Center Loss
+    """
+    def __init__(self, num_classes, feature_size, ctx, alpha=0.5, lmbd=0.01, weight=1., batch_axis=0, **kwargs):
+        super(CenterLoss, self).__init__(weight*lmbd, batch_axis, **kwargs)
+        self._alpha = alpha
+        self._lmbd = lmbd
+        self._num_classes = num_classes
+        self._feature_size = feature_size
+        self._ctx = ctx
+        self._centers = nd.zeros((num_classes, feature_size), ctx=ctx)
+
+    def hybrid_forward(self, F, output, label, sample_weight=None):
+        selected_centers = F.take(self._centers, label)
+        diff = output - selected_centers
+        loss = F.sum(F.square(diff), axis=1)
+
+        loss = _apply_weighting(F, loss, self._weight, sample_weight)
+        # Update the centers
+        counts = nd.zeros((self._num_classes, ), ctx=self._ctx)
+        for lb in label:
+            lb = int(lb.asscalar())
```

Comment:
These are bad practice. We should add a bincount operator to do this

Reply:
Will change.